### PR TITLE
Chore: Fix dashboards label adding workflow

### PR DESCRIPTION
**Pull request by mdvictor**
**What is this feature?**

Fixes the workflow by using a github app generated token instead of a access token.

**Why do we need this feature?**

Because the workflow isn't running anymore

**Who is this feature for?**

Dashboards squad

### CODE REVIEW

File: .github/workflows/dashboards-issue-add-label.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ on:
     types: [opened, closed, edited, reopened, assigned, unassigned, labeled, unlabeled]
```
Comment (academo) [inline .github/workflows/dashboards-issue-add-label.yml:5]: You need to add permissions to this workflow to be able to pull the secrets from vault
```suggestion
permissions:
  contents: read
  id-token: write
```

File: .github/workflows/dashboards-issue-add-label.yml [hunk 2]
```diff
@@ -22,7 +21,7 @@ jobs:
         id: check
         shell: bash
         run: |
-          if [ -n "${{ (secrets.ISSUE_COMMANDS_TOKEN != '') || '' }}" ]; then
+          if [ "${{ github.repository }}" == "grafana/grafana" ] && [ -n "${{ secrets.GRAFANA_MISC_STATS_API_KEY }}" ]; then
```
Comment (academo) [inline .github/workflows/dashboards-issue-add-label.yml:24]: do you really need `GRAFANA_MISC_STATS_API_KEY`?

Comment (mdvictor) [reply .github/workflows/dashboards-issue-add-label.yml]: No, I don't. And yeah, that sounds good. I just removed the entire job. Thank you!

**Comment by mdvictor** [comment]
Tested this against `github-actions-testrepo` [here](https://github.com/grafana/github-actions-testrepo/blob/main/.github/workflows/dashboards-issue-add-label.yml), works like a charm, thanks @academo !

**Review by axelavargas — APPROVED** [review]
🥳 great work, @mdvictor

**Pull request merged**